### PR TITLE
feature-bnb

### PR DESCRIPTION
**Pull request by BitHighlander**
(WIP)

# Binance Chain support

This pr enables the transfer transaction type for the BNB mainnet chain. 

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

Steps to test:

Targeted kk firmware release: 6.4.1 
(support currently is in master firmware)

Notes:
This PR supports ONLY transfer messages

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     ]
   },
   "scripts": {
+    "yarn": "yarn",
```
Comment (SteveHadow) [inline package.json]: I don't think we need this(?)

Comment (BitHighlander) [reply package.json]: we don't *not need this. Yarn is required in the flow and I prefer to document it.

**Comment by SteveHadow** [comment]
Any integration tests?

**Comment by elmutt** [comment]
hard to see sandbox changes that are relavent because of formatting changes

File: examples/sandbox/index.ts [hunk 2]
```diff
@@ -433,6 +476,85 @@ $getAppInfo.on('click', async (e) => {
...
+ * Binance
+ */
+const $binanceAddr = $('#binanceAddr')
+const $binanceTx = $('#binanceTx')
+const $binanceResults = $('#binanceResults')
+
+$binanceAddr.on('click', async (e) => {
+  e.preventDefault()
+  if (!wallet) {
+    $ethResults.val("No wallet?");
```
Comment (kaladinlight) [inline examples/sandbox/index.ts]:
```suggestion
    $binanceAddr.val("No wallet?");
```

File: examples/sandbox/index.ts [hunk 3]
```diff
@@ -433,6 +476,85 @@ $getAppInfo.on('click', async (e) => {
...
+  } else {
+    let label = await wallet.getLabel()
+    $binanceResults.val(label + " does not support Binance")
+  }
+})
+
+$binanceTx.on('click', async (e) => {
+  e.preventDefault()
+  if (!wallet) {
+    $ethResults.val("No wallet?");
```
Comment (kaladinlight) [inline examples/sandbox/index.ts]:
```suggestion
    $binanceResults.val("No wallet?");
```

File: package.json (hunk 1 above)
Comment (kaladinlight) [reply package.json]: maybe document in a README? Agreed, seems weird to put something like that here...

File: packages/hdwallet-core/src/binance.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+export interface BinanceSignTx {
+  addressNList: BIP32Path
+  tx: BinanceTx
+  chain_id: string
+  account_number: string
+  sequence: string
+}
+
+
+export type BinanceSignedTx = BinanceTx
```
Comment (kaladinlight) [inline packages/hdwallet-core/src/binance.ts]: If signatures are only used for `BinanceSignedTx` maybe remove from `BinanceTx` and use an intersection to create the signed tx interface?
```
export interface BinanceSignedTx = BinanceTx && {
  signatures: {
    pub_key: string
    signature: string
  }
}
```

File: packages/hdwallet-keepkey/src/keepkey.ts [hunk 5]
```diff
@@ -257,10 +264,51 @@ function describeCosmosPath (path: BIP32Path): PathDescription {
   }
 }
 
-export class KeepKeyHDWalletInfo implements HDWalletInfo, BTCWalletInfo, ETHWalletInfo, CosmosWalletInfo {
+function describeBinancePath (path: BIP32Path): PathDescription {
+  let pathStr = addressNListToBIP32(path)
+  let unknown: PathDescription = {
+    verbose: pathStr,
+    coin: 'Atom',
```
Comment (kaladinlight) [inline packages/hdwallet-keepkey/src/keepkey.ts]:
```suggestion
    coin: 'Binance',
```

File: package.json (hunk 1 above)
Comment (SteveHadow) [reply package.json]: It just seems redundant. Why do we need a script called `yarn` that just runs `yarn` lol puts it in the README that running `yarn` is required for the project to work.

File: packages/hdwallet-core/src/binance.ts (hunk 4 above)
Comment (BitHighlander) [reply packages/hdwallet-core/src/binance.ts]: :thinking:

File: examples/sandbox/index.ts [hunk 6]
```diff
@@ -433,6 +476,85 @@ $getAppInfo.on('click', async (e) => {
...
+})
+
+$binanceTx.on('click', async (e) => {
+  e.preventDefault()
+  if (!wallet) {
+    $binanceResults.val("No wallet?");
+    return
+  }
+  if (supportsBinance(wallet)) {
+    console.log("checkpoint1")
```
Comment (mcchadwick) [inline examples/sandbox/index.ts:517]: There's a debug print statement left in here.

**Pull request merged**